x/gov/genesis.go

```diff
@@ -13,6 +13,32 @@ const (
 	DefaultPeriod time.Duration = 86400 * 2 * time.Second // 2 days
 )
 
+var (
+	startingProposalID uint64
+)
+
+// MsgVote.ValidateBasic() needs StartingProposalID provided by the genesis
+// but it cannot know about the genesis / keeper information
+// setStartingProposalID ensures setting startingProposalID only once
+// so unless there is more than one gov keeper the var will store the genesis StartingProposalID
+// TODO: deactivated for testing
+func setStartingProposalID(id uint64) {
+	/*
+		if initedStartingProposalID {
+			panic("cannot set multiple genesis StartingProposalID")
+		}
+		onceStartingProposalID.Do(func() {
+			startingProposalID = id
+			initedStartingProposalID = true
+		})
+	*/
```

Comment:
should this commented code be here still or deleted? if it shouldn't be deleted what do we need to do before uncommenting it? (should add that information in a comment)

Reply:
It is needed because of https://github.com/cosmos/cosmos-sdk/pull/3880#discussion_r266869513, but I'm not sure if it is really needed to prevent depositing/voting on the proposals before the startingProposalID. For example, a remaining proposal can be on either depositperiod or votingperiod on the new chain too, which means that we allow tho deposit/vote on it.

Reply:
Removed the logic https://github.com/cosmos/cosmos-sdk/pull/3880/commits/dd0e47606cf2a17d7bc580865aa94d09e26b2d16